pkgs/development/python-modules/screeninfo/default.nix

```diff
@@ -0,0 +1,27 @@
+{ stdenv, buildPythonApplication, fetchPypi, isPy36, dataclasses, libX11, libXinerama, libXrandr }:
+
+buildPythonApplication rec {
+  pname = "screeninfo";
+  version = "0.6.5";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "0vcw54crdgmbzwlrfg80kd1a8p9i10yks8k0szzi0k5q80zhp8xz";
+  };
+
+  # dataclasses is a compatibility shim for python 3.6 ONLY
+  patchPhase = if isPy36 then "" else ''
+    substituteInPlace setup.py \
+      --replace "\"dataclasses\"," ""
+  '' + ''
+    substituteInPlace screeninfo/enumerators/xinerama.py \
+      --replace "load_library(\"X11\")" "ctypes.cdll.LoadLibrary(\"${libX11}/lib/libX11.so\")" \
+      --replace "load_library(\"Xinerama\")" "ctypes.cdll.LoadLibrary(\"${libXinerama}/lib/libXinerama.so\")"
+    substituteInPlace screeninfo/enumerators/xrandr.py \
+      --replace "load_library(\"X11\")" "ctypes.cdll.LoadLibrary(\"${libX11}/lib/libX11.so\")" \
+      --replace "load_library(\"Xrandr\")" "ctypes.cdll.LoadLibrary(\"${libXrandr}/lib/libXrandr.so\")"
+  '';
```

Comment:
`load_library` is defined in `utils.py`, and is basically just a wrapper around

```py
path = ctypes.util.find_library(name)
return ctypes.cdll.LoadLibrary(path)
```

On Linux, this uses `dlopen`, so if you just `wrapProgram` the binaries to add `libX11` and `libXrandr`'s lib path to `LD_LIBRARY_PATH`, these libraries should be found even without patching the source.

Reply:
I do not like using `LD_LIBRARY_PATH` since it can lead to weird and difficult to diagnose error when it loads a different version of library than the program is compiled against. Remember environment variables are inherited by child processes and `LD_LIBRARY_PATH` has higher priority than `DT_RUNPATH` ELF entry we use to locate libraries.

For a concrete example when this occurs: If you have a URL hyperlink in the application and click it, it will open a browser child process, which will inherit these X libraries. If you use browser from a different channel, it will likely crash.

Reply:
Fair enough - Can we still add the two library paths into `utils.py`, instead of hunting for each and every occurence?

Reply:
I could patch the `load_library` function in `utils.py`, but it felt neater to me to just use `substituteInPlace` as those libraries are highly unlikely to appear anywhere except in the files which get patched